modules/product/src/main/java/com/opengamma/strata/product/bond/BillTrade.java

```diff
@@ -0,0 +1,583 @@
+import com.opengamma.strata.product.ProductType;
+import com.opengamma.strata.product.ResolvableTrade;
+import com.opengamma.strata.product.SecuritizedProductTrade;
+import com.opengamma.strata.product.TradeInfo;
+import com.opengamma.strata.product.common.SummarizerUtils;
+
+import java.util.Map;
+import java.util.NoSuchElementException;
+import org.joda.beans.Bean;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaBean;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+/**
+ * A trade representing a bill.
+ * <p>
+ * A trade in an underlying {@link Bill}.
+ * 
+ * <h4>Price and yield</h4>
+ * Strata uses <i>decimal</i> yields and prices for bills in the trade model, pricers and market data.
+ * For example, a price of 99.32% is represented in Strata by 0.9932 and a yield of 1.32% is represented by 0.0132.
+ */
+@BeanDefinition(constructorScope = "package")
+public final class BillTrade
+    implements SecuritizedProductTrade<Bill>, ResolvableTrade<ResolvedBillTrade>, ImmutableBean, Serializable {
+
+  /**
+   * The additional trade information, defaulted to an empty instance.
+   * <p>
+   * This allows additional information to be attached to the trade.
+   */
+  @PropertyDefinition(validate = "notNull", overrideGet = true)
+  private final TradeInfo info;
+  /**
+   * The bill that was traded.
+   * <p>
+   * The product captures the contracted financial details of the trade.
+   */
+  @PropertyDefinition(validate = "notNull", overrideGet = true)
+  private final Bill product;
+  /**
+   * The quantity that was traded.
+   * <p>
+   * This will be positive if buying and negative if selling.
+   */
+  @PropertyDefinition(overrideGet = true)
+  private final double quantity;
+  /**
+   * The price at which the bill was traded, in decimal form.
+   * <p>
+   * One of the traded yield and price should be provided.
+   */
+  @PropertyDefinition(validate = "ArgChecker.notNegativeOrZero", overrideGet = true)
+  private final double price;
+
+  @ImmutablePreBuild
+  private static void preBuild(Builder builder) {
+    ArgChecker.isTrue(builder.info.getSettlementDate().isPresent()
+        || builder.info.getTradeDate().isPresent(), "Bill trades need a settlement date or trade date.");
+  }
+  
+  /**
+   * Generates a Bill trade instance where the price is computed from the traded yield.
+   * 
+   * @param info  the additional trade information
+   * @param product  the bill that was traded
+   * @param quantity  the quantity that was traded
+   * @param yield  the yield at which the bill was traded
+   * @return  the instance
+   */
+  public static BillTrade ofYield(TradeInfo info, Bill product, double quantity, double yield) {
+    ArgChecker.isTrue(info.getSettlementDate().isPresent(),
```

Comment:
Could use same logic to calculate the settlement date from the trade date.

Reply:
No reference data available here.